docs/arcadia-users-group/20221011-developer-productivity/lesson.md

```diff
@@ -0,0 +1,172 @@
+
+## Accessing the shell for this lesson
+
+For this lesson, we need to have access to a Unix shell.
+If you're not sure how to open a terminal on your computer, see [these instructions](https://swcarpentry.github.io/shell-novice/setup.html).
+
+## Caveats
+
+- You don't really have to do any of these customizations to get good at programming. But these are designed to make your life easier.
+- A good chunk of these customizations are personal. The goal is to learn _how_ to customize and to empower you to customize as you wish.
+- A good chunk of these customizations are stylistic. As a programmar, you spend a lot of time in your terminal or text editor. I think it's important to make these tools beautiful.
+- I like bash because I've been using it for 10 years. This workshop will focus on using bash instead of zsh. But there are parallels between the two and learning to customize one will help you customize the other.
+- These type of tools can sometimes get RAM-intensive. So, beware if you have a laptop with <16GB of RAM.
+
+## Demo repository
+
+For demo purposes, we'll use the contents of the [AUG repository](https://github.com/Arcadia-Science/arcadia-computational-training). So, let's download the contents of it:
+
+```{bash}
+cd ~/Desktop
+git clone git@github.com:Arcadia-Science/arcadia-computational-training.git
+cd arcadia-computational-training
+ls
+```
+
+## IDEs
+
+### What are IDEs?
+
+People think of programming as mostly editing the source code of a web application or a script or a computational pipeline. But there are many other aspects of programming that happen all the time that we take for granted: debugging, testing, checking the code for errors, typos etc.
+
+Integrated development environments or IDEs bring all these aspects of programming together and offer convenient features like autocomplete, syntax highlighting, linting, code execution etc.
+
+IDEs are not necessary for programming, but with basic configuration, they can significantly increase coding productivity.
+
+### Which IDE should I use?
+
+There are many IDEs or IDE-like text editors out there. Some are language-specific (think [IntelliJ](https://www.jetbrains.com/idea/) for Java). Some are generic like [Sublime Text](https://www.sublimehq.com/). Up until very recently, I've exclusively used Sublime Text and still love it for light-weight programming.
+
+But at Arcadia Science, we'll be using [Visual Studio Code](https://code.visualstudio.com/) (aka VS Code). There are couple reasons for this:
+
+- It has a tight-integration with git and Github.
+- It has a rich marketplace of free extensions, enabling you to configure your IDE to your heart's content.
+- It works cross-platform and has a straightforward way of syncing your settings across devices.
+- It is quite easy to configure because it suggests improvements on its own as needed.
+- It has access to a built-in terminal. Try hitting CTRL + ` when you install it.
```

Comment:
This backtick is going to mess with the formatting on the rest of the page. I'm not sure how to escape it...